File: set_result.go
// Random Go struct generated on 2025-06-03
package main

import (
    "encoding/json"
    "fmt"
    "time"
)

type HandleValue struct {
    Name      string    `json:"name"`
    Value     int       `json:"value"`
    Active    bool      `json:"active"`
    CreatedAt time.Time `json:"created_at"`
}

func NewHandleValue(name string, value int) *HandleValue {
    return &HandleValue{
        Name:      name,
        Value:     value,
        Active:    true,
        CreatedAt: time.Now(),
    }
}

func (s *HandleValue) UpdateValue(newValue int) {
    s.Value = newValue
}

func (s *HandleValue) ToJSON() ([]byte, error) {
    return json.Marshal(s)
}

func main() {
    obj := NewHandleValue("sample", 67)
    data, _ := obj.ToJSON()
    fmt.Println(string(data))
}

File: set_result.rb
# Random Ruby class generated on 2025-09-18

class CheckResult
  attr_reader :name, :value, :created_at
  attr_writer :name, :value

  def initialize(name, value = 67)
    @name = name
    @value = value
    @created_at = Time.now
  end

  def to_hash
    {
      name: @name,
      value: @value,
      created_at: @created_at.iso8601
    }
  end

  def update_value(new_value)
    old_value = @value
    @value = new_value
    "Updated from #{old_value} to #{new_value}"
  end
end

# Example usage
obj = CheckResult.new("sample", 34)
puts obj.to_hash
